integration-tests/bytecode-enhancements-it/src/test/java/org/hibernate/reactive/it/LazyBasicFieldTest.java

```diff
@@ -47,4 +50,46 @@ public void testFetchBasicField(VertxTestContext context) {
 								) ) )
 		);
 	}
+
+	@Test
+	public void testTransparentLazyFetching(VertxTestContext context) {
+		final Crew emily = new Crew();
+		emily.setId( 21L );
+		emily.setName( "Emily Jackson" );
+		emily.setRole( "Passenger" );
+		emily.setFate( "Unknown" );
+
+		test( context, assertThrown( LazyInitializationException.class, getMutinySessionFactory()
+					  .withTransaction( session -> session.persist( emily ) )
+					  .call( () -> getMutinySessionFactory().withSession( session -> session.find( Crew.class, emily.getId() )
+							  .invoke( Crew::getRole ) ) )
+			  ).invoke( exception -> assertThat( exception.getMessage() ).contains( "Reactive sessions do not support transparent lazy fetching" ) )
+		);
+	}
+
+	@Test
+	public void testGetReferenceAndTransparentLazyFetching(VertxTestContext context) {
+		final Crew emily = new Crew();
+		emily.setId( 21L );
+		emily.setName( "Emily Jackson" );
+		emily.setRole( "Passenger" );
+		emily.setFate( "Unknown" );
+
+		test( context, assertThrown( LazyInitializationException.class, getMutinySessionFactory()
+					  .withTransaction( session -> session.persist( emily ) )
+					  .chain( () -> getMutinySessionFactory().withSession( session -> {
+						  Crew crew = session.getReference( Crew.class, emily.getId() );
+						  String role = crew.getRole();
+						  return session.flush();
+					  } ) )
+			  ).invoke( exception -> assertThat( exception.getMessage() ).contains( "Reactive sessions do not support transparent lazy fetching" ) )
+		);
+	}
+
+	public static <U extends Throwable> Uni<U> assertThrown(Class<U> expectedException, Uni<?> uni) {
```

Comment:
Don't we already have this in [ReactiveAssertions#assertThrown](https://github.com/hibernate/hibernate-reactive/blob/82340d1dfb1ae6c074cae226c2ff04c17b25b40b/hibernate-reactive-core/src/test/java/org/hibernate/reactive/testing/ReactiveAssertions.java#L65)?

Reply:
maybe we can copy it from `hibernate-reactive-core`